src/cross_section/src/cross_section.cpp

```diff
@@ -236,21 +237,30 @@ C2::PathsD CrossSection::GetPaths() const {
  * @param center Set to true to shift the center to the origin.
  */
 CrossSection CrossSection::Square(const glm::vec2 dims, bool center) {
+  if (glm::length(dims) == 0.0f) {
+    return CrossSection();
+  }
+
+  // reverse winding if dimensions are negative
```

Comment:
Considering we're marking negative radius as invalid, should we just mark negative dimensions invalid too? Likewise for Manifold of course.

Reply:
That would also make sense,  for reference, negative dimensions for squares and cubes results in empty shapes in OpenSCAD. On the other hand at least negative dimensions of squares/cubes have some intuitive sense/purpose (like negative scaling does),compared to negative radius which seems more obviously not useful (only interpretation other than failing is ignoring it).

Reply:
Yeah, let's make that change.

Reply:
Ok, I'll add a negative (on any dimension) check for square and cube then.